packages/odyssey-react-mui/src/labs/FileUpload.tsx

```
@@ -0,0 +1,294 @@
+  /**
+   * an array of file types the user is able to upload. @see https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/accept#unique_file_type_specifiers for examples
+   */
+  acceptedFileTypes?: string[];
+  /**
+   * The label for the `input` element.
+   */
+  label: string;
+  /**
+   * Function that is called when the list of ifles to upload is changed
+   */
+  onChange: (files: File[]) => void;
+  /**
+   * Either `single` or `multiple`. If `multiple`, multiple files can be uploaded
+   */
+  type?: (typeof fileUploadTypes)[number];
+  /**
+   * Either `button` or ``. If `multiple`, multiple files can be uploaded
+   */
+  variant: (typeof fileUploadVariants)[number];
+} & Pick<
+  FieldComponentProps,
+  | "errorMessage"
+  | "hint"
+  | "HintLinkComponent"
+  | "id"
+  | "isDisabled"
+  | "isOptional"
+>;
+
+type RemoveFileFromFilesToUploadList = (name: string) => void;
+
+const FileUpload = ({
+  acceptedFileTypes,
+  errorMessage,
+  id,
+  isDisabled = false,
+  isOptional,
+  hint,
+  HintLinkComponent,
+  label,
+  onChange,
+  type,
+  variant,
+}: FileUploadProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+  const { t } = useTranslation();
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [filesToUpload, setFilesToUpload] = useState<File[]>([]);
+
+  useEffect(() => {
+    onChange(filesToUpload);
+  }, [filesToUpload, onChange]);
+
+  const updateFilesToUpload = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const { files } = event.target;
+
+      if (files && files.length > 0) {
+        const mergedFiles =
+          type === "multiple" ? [...filesToUpload, ...files] : [...files];
+
+        setFilesToUpload(mergedFiles);
+      }
+
+      // reset input value to allow re-upload of a file with the same name
+      event.target.value = "";
+    },
+    [type, filesToUpload],
+  );
+
+  const triggerFileInputClick = useCallback(() => {
+    inputRef?.current?.click();
```

`inputRef?.current` shouldn't need a question mark:
```suggestion
    inputRef.current?.click();
```

---

Probably not but, don't think it really hurst anything being there either though. Just extra sure it won't fail :)

---

I'd rather be explicit about our intentions. It's something called police coding or helicopter ~parenting~ coding.

We're basically trying to address issues that don't exist, and it will hide bugs. We _want_ it to error if `inputRef.current` doesn't exist because React would be broken in that case. Right now, we'd be masking that error.